src/sys/windows.rs

```diff
@@ -774,6 +775,53 @@ pub(crate) fn to_mreqn(
     }
 }
 
+#[allow(unused_unsafe)] // TODO: replace with `unsafe_op_in_unsafe_fn` once stable.
+pub(crate) fn unix_sockaddr(path: &Path) -> io::Result<SockAddr> {
+    unsafe {
+        SockAddr::try_init(|storage, len| {
```

Comment:
We've change our `SockAddr` constructors usage a bit since this was written, can you change it to use `SockAddr::new`, e.g: https://github.com/rust-lang/socket2/blob/eaa6300e13d7ed421a3df2fe0149b7a54956b84c/src/sys/unix.rs#L540-L552

Reply:
I was modeling this after the implementation in `unix.rs`. Would you like me to change both?

Reply:
Yes please.